exampleCourse/questions/demo/annotated/engCircuit/server.py

```diff
@@ -1,40 +1,93 @@
 import random
 
+from schemdraw import Drawing
+import schemdraw.elements as elm
 
-def generate(data):
-    ask = ["equivalent resistance $R_T$", "current from the power supply $I_T$"]
-    which = random.choice([0, 1])
-    data["params"]["ask"] = ask[which]
+def file(data):
+    if data["filename"] != "figure.svg":
+        return
+    
+    drawing = Drawing()
+    
+    R1 = str(data["params"]["R1"]) + r" $\Omega$"
+    R2 = str(data["params"]["R2"]) + r" $\Omega$"
+    R3 = str(data["params"]["R3"]) + r" $\Omega$"
+    Vt = str(data["params"]["Vt"]) + " V"
+
+    match data["params"]["whichfig"]:
+        case "R_T": # variant: Find total resistance
+
+            # drawing the circuit using schemdraw
+            drawing.push()
+            drawing += elm.Line()
+            drawing.push()
+            drawing += elm.Resistor().down().label(R1)
+            drawing.pop()
+            drawing += elm.Line().right()
+            drawing.push()
+            drawing += elm.Resistor().down().label(R2)
+            drawing.pop()
+            drawing += elm.Line().right()
+            drawing += elm.Resistor().down().label(R3)
+            drawing += elm.Line().left()
+            drawing += elm.Line().left()
+            drawing += elm.Line().left()
+            drawing.pop()
+            drawing += elm.BatteryCell().down().label(Vt,'bottom')
 
-    label = ["$R_T$", "$I_T$"]
-    data["params"]["lab"] = label[which]
+        case "I_T": # variant: Find current
 
-    unit = ["$\\Omega$", "A"]
-    data["params"]["unit"] = unit[which]
+            # drawing the circuit using schemdraw
+            drawing.push()
+            drawing += elm.Line().right()
+            drawing += elm.Line().right()
+            drawing += elm.Line().right()
+            drawing += elm.Line().down()
+            drawing += elm.Resistor().left().label(R3)
+            drawing += elm.Resistor().left().label(R2)
+            drawing += elm.Resistor().left().label(R1)
+            drawing.pop()
+            drawing += elm.BatteryCell().down().label(Vt,'bottom')
 
+    return drawing.get_imagedata()
+
+
+def generate(data):
+
+    # Randomly choose Vt, R1, R2, R3
     Vt = random.randint(100, 200)
     data["params"]["Vt"] = Vt
 
     R1 = random.choice(list(range(20, 180, 10)))
     data["params"]["R1"] = R1
-
+    
     R2 = random.choice(list(range(20, 180, 20)))
     data["params"]["R2"] = R2
-
+    
     R3 = random.choice(list(range(20, 100, 5)))
```

Comment:
If you're switching to a units-input element, you can use `pint` to add units to each of these generated quantities. Then it's not necessary to explicitly set the unit in `data["params"]`. See [the pint documentation](https://pint.readthedocs.io/en/0.22/getting/tutorial.html) for more information.

Reply:
Simplified code is here: https://github.com/PrairieLearn/PrairieLearn/pull/8179/commits/e9d7d5ac1bcf8fca62c502f618e38d30042648db